R/post-body.R

```diff
@@ -24,3 +34,33 @@ parseBody <- function(body){
   }
   ret
 }
+
+
+#' @importFrom utils URLdecode
+#' @noRd
+parseMultipart <- function(body, boundary){
+  function(val, req, res, errorHandler){
+    tryCatch({
+      if (!requireNamespace("webutils", quietly = TRUE)) {
+        stop("The webutils package is not available but is required in order
+          to use the functionality to POST binary files",
+             call. = FALSE)
+      }
+      # Is there data in the request?
+      if (is.null(body) || length(body) == 0 || body == "") {
+        return(list())
+      }
+      parsed_binary <- webutils::parse_multipart(body, boundary)
+      file_name <- parsed_binary$myfile$filename
+      file_data <- parsed_binary$myfile$value
+
+      tmpfile <- tempfile()
```

Comment:
Hm. It's unfortunate that we're not able to clean up these temp files when we're done. But if we added an `on.exit` here the file would be gone before we could access the file downstream. Maybe I need to add a hook to the response object so that we can clean up any associated resources when it's done.